controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -209,6 +205,44 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		rosaScope.ControlPlane.Status.FailureMessage = nil
 	}
 
+	if errs := rosaScope.ControlPlane.Spec.AdditionalTags.Validate(); errs != nil {
+		var msg []string
+		for _, err := range errs {
+			msg = append(msg, err.Error())
+		}
+		rosaScope.ControlPlane.Status.FailureMessage = ptr.To(strings.Join(msg, ", "))
+		// dont' requeue because input is invalid and manual intervention is needed.
+		return ctrl.Result{}, nil
+	}
```

Comment:
why outside of `validateControlPlaneSpec()` func? I think we should encapsulate all validations in one place.

Reply:
It gets muddy with the CIDRs since having the validation for them inside of the func means we need to pass the parsed values back out. Are you OK with having the tag validation inside, but the CIDR outside?